core/primitives/src/transaction.rs

```diff
@@ -861,6 +862,24 @@ impl From<ReceiptTransaction> for receipt_proto::ReceiptTransaction {
     }
 }
 
+impl Hash for ReceiptTransaction {
```

Comment:
Shouldn't has of ReceiptTransaction be the hash of transaction?

Reply:
Yeah I implemented this to put receipts in `HashSet`, but actually I can just use a hashmap.